lib/factory.rb

```diff
@@ -14,3 +13,86 @@
 # - to_a
 # - values_at
 # - ==, eql?
+class Factory
+  class << self
+    def new(*args, &block)
+      const_set(args.shift.capitalize, class_new(*args, &block)) if args.first.is_a?(String)
+      class_new(*args, &block)
```

Comment:
i think `create_class` is a better name for the method
method names are ususally verbs: `create`, `update`, `validate` etc.